src/Dapr.Actors/Runtime/ConverterUtils.cs

```diff
@@ -60,5 +63,69 @@ public static string ConvertTimeSpanValueInDaprFormat(TimeSpan? value)
 
             return stringValue;
         }
+
+        public static string ConvertTimeSpanValueInISO8601Format(TimeSpan value, int repetitions)
+        {
+            if (repetitions <= 0)
+            {
+                return ConvertTimeSpanValueInDaprFormat(value);
+            }
+
+            StringBuilder builder = new StringBuilder();
+            builder.AppendFormat("R{0}/P", repetitions);
+
+            if(value.Days > 0)
+            {
+                builder.AppendFormat("{0}D", value.Days);
+            }
+
+            builder.Append("T");
+
+            if(value.Hours > 0)
+            {
+                builder.AppendFormat("{0}H", value.Hours);
+            }
+
+            if(value.Minutes > 0)
+            {
+                builder.AppendFormat("{0}M", value.Minutes);
+            }
+
+            if(value.Seconds > 0)
+            {
+                builder.AppendFormat("{0}S", value.Seconds);
+            }
+            return builder.ToString();
+        }
+        
+        public static (TimeSpan, int) ConvertTimeSpanValueFromISO8601Format(string valueString)
```

Comment:
If this is used needed for test code then this **must** live in the tests. Regexes in production code need to be scrutinized **heavily** for security reasons.

Reply:
nevermind, I see where this is used in the product code path. I want to see if we can find a solution to this that doesn't involve regexes if we can avoid it.

Reply:
@layomia - do you know if S.T.J or other parts of the runtime provide anything here that will help? I don't want to introduce new parsers or regexes that I don't trust 😆

Reply:
The ISO 8601 duration format doesn't actually map directly to our .NET `TimeSpan` type. `TimeSpan` is an absolute representation in that it counts number of ticks, while ISO duration is relative - allowing representations like "day", and "month". That's likely why it isn't included as part of the [standard formats for `TimeSpan`](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-timespan-format-strings).

STJ recently (and finally! 😄)  added support for `TimeSpan` (https://github.com/dotnet/runtime/pull/54186), but that supports only the "standard .NET" (`c`) format. So AFAIK, there are no built-in types to support this lossy conversion from ISO 8601 duration to .NET `TimeSpan`. We'd need a new type in the BCL to support the ISO format. This leaves us with custom parsers like with this PR.